### PR TITLE
Fix incorrect edit comment countdown timer calculation #74

### PR DESCRIPTION
**Pull request by jackburg**

### CODE REVIEW

**Comment by jackburg** [comment]
I've re-forked repo & reapply my changes, still no luck, please advice how to resolve this issue.

**Comment by umputun** [comment]
This is the reason 
https://docs.travis-ci.com/user/environment-variables/#Defining-Variables-in-Repository-Settings

> ...  we do not provide these values to untrusted builds, triggered by pull requests from another repository.

Looks like PR build can't access such variables. I'll see how to deal with this. Will try to eliminate the need of external mongo for tests or (as plan B) skip mongo-related tests in PR builds

**Comment by umputun** [comment]
Changed (in master) the way travis runs mongo-related tests. Hopefully, it will resolve the issue you had. Just merge master into your PR and it should pass

File: backend/app/main_test.go [hunk 1]
```diff
@@ -73,6 +75,60 @@ func TestApplicationDevMode(t *testing.T) {
...
+		t.Skip("skip mongo app test")
+	}
+
+	opts := Opts{}
+	// prepare options
+	p := flags.NewParser(&opts, flags.Default)
+	_, err := p.ParseArgs([]string{"--secret=123456", "--dev-passwd=password", "--url=https://demo.remark42.com",
+		"--cache.type=mongo", "--store.type=mongo", "--avatar.type=mongo", "--mongo.url=" + mongoURL, "--mongo.db=test_remark", "--port=12345"})
+	require.Nil(t, err)
+	opts.Auth.Github.CSEC, opts.Auth.Github.CID = "csec", "cid"
```
Comment (golangci) [inline backend/app/main_test.go]: string `csec` has 5 occurrences, make it a constant

File: backend/app/main_test.go [hunk 2]
```diff
@@ -73,6 +75,60 @@ func TestApplicationDevMode(t *testing.T) {
...
+	opts.BackupLocation = "/tmp"
```
Comment (golangci) [inline backend/app/main_test.go]: string `/tmp` has 4 occurrences, make it a constant

File: backend/app/rest/cache/mongo_test.go [hunk 3]
```diff
@@ -0,0 +1,312 @@
...
+	assert.Equal(t, "result-3", string(res), "should be cached")
+
+	// try to cache after maxKeys reached
+	res, err = lc.Get(NewKey("site").ID("key-X"), func() ([]byte, error) {
+		return []byte("result-X"), nil
+	})
+	assert.Nil(t, err)
+	assert.Equal(t, "result-X", string(res))
+
+	conn.WithCustomCollection("cache", func(coll *mgo.Collection) error {
```
Comment (golangci) [inline backend/app/rest/cache/mongo_test.go]: Error return value of `conn.WithCustomCollection` is not checked

File: backend/app/rest/cache/mongo_test.go [hunk 4]
```diff
@@ -0,0 +1,312 @@
...
+	assert.Nil(t, err)
+	assert.Equal(t, "result-Z", string(res))
+
+	res, err = lc.Get(NewKey("site").ID("key-Z"), func() ([]byte, error) {
+		return []byte("result-Zzzz"), nil
+	})
+	assert.Nil(t, err)
+	assert.Equal(t, "result-Z", string(res), "got cached value")
+
+	conn.WithCustomCollection("cache", func(coll *mgo.Collection) error {
```
Comment (golangci) [inline backend/app/rest/cache/mongo_test.go]: Error return value of `conn.WithCustomCollection` is not checked

File: backend/app/rest/cache/mongo_test.go [hunk 5]
```diff
@@ -0,0 +1,312 @@
...
+	res, err := lc.Get(NewKey("site").ID("key").Scopes("s1", "s2"), func() ([]byte, error) {
+		return []byte(data), nil
+	})
+	require.Nil(b, err)
+	require.True(b, strings.HasPrefix(string(res), "xxxx"), string(res))
+
+	key := NewKey("site").ID("key").Scopes("s1", "s2")
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		lc.Get(key, func() ([]byte, error) {
```
Comment (golangci) [inline backend/app/rest/cache/mongo_test.go]: Error return value of `lc.Get` is not checked

File: backend/app/rest/cache/mongo_test.go [hunk 6]
```diff
@@ -0,0 +1,312 @@
...
+			b.Fail()
+			return nil, nil
+		})
+	}
+}
+
+func mongoCacheSize(t *testing.T, conn *mongo.Connection) (count int) {
+	conn.WithCustomCollection("cache", func(coll *mgo.Collection) (e error) {
```
Comment (golangci) [inline backend/app/rest/cache/mongo_test.go]: Error return value of `conn.WithCustomCollection` is not checked

**Comment by jackburg** [comment]
Yay! Thank you @umputun!

**Comment by jackburg** [comment]
Looks like we should disable go-lint for your mongo test file as well.

**Comment by Guria** [comment]
Seems you try to develop in master and constantly pull updates from the remote.
CI sees this updates as new commits and prevents normal execution

**Pull request closed without merging**